random-referral/app/controllers/referrals_controller.rb

```diff
@@ -1,5 +1,12 @@
 class ReferralsController < ApplicationController
   def show
+    @referral = Referral.all.find do |referral|
+      referral.id == params[:id]
```

Comment:
Do we need to gracefully handle a case where the referral ID is not found?

Reply:
Do you mean to render an error message if ID is not found?

Reply:
Yes, I'm not sure what will happen if the ID is not found - will it show a generic Rails not found error? Do we want it to show a custom error page? I don't think it's hugely important however.